x-pack/plugins/spaces/public/ui_api/lazy_wrapper.tsx

```diff
@@ -19,22 +20,18 @@ interface InternalProps<T> {
   props: JSX.IntrinsicAttributes & PropsWithRef<PropsWithChildren<T>>;
 }
 
-export const LazyWrapper: <T>(props: InternalProps<T>) => ReactElement = ({
+export const LazyWrapper: <T>(props: InternalProps<T>) => ReactElement | null = ({
   fn,
   getStartServices,
   props,
 }) => {
-  const [notifications, setNotifications] = useState<NotificationsStart | undefined>(undefined);
-  useEffect(() => {
-    getStartServices().then(([coreStart]) => {
-      setNotifications(coreStart.notifications);
-    });
-  });
+  const { value: startServices = [{ notifications: undefined }] } = useAsync(getStartServices);
```

Comment:
For next time: An empty object should be enough here:
```
const { value: startServices = [{}] } = useAsync(getStartServices);
```

Reply:
Ah, I did this because there's no type overlap between `CoreStart` and `{}`, so tsc complains about that. Using `{ notifications: undefined }` allows for type inferencing. Now that you mention it though I suppose this would work
```ts
const { value: startServices = [{} as CoreStart] } = useAsync(getStartServices);
```
I think it's a bit better to have explicit types but we've got a big mix of inferences + explicit typing everywhere 🙈